Dashboard enhancement

kimnikitaschijf opened the pull request:
## What does this change?

Resolves issue #176

Verbeteringen aan het dashboard om overzicht en gebruiksvriendelijkheid te verhogen.  
- Progress label-kleur aangepast  
- Layout “Assessor 2” verbeterd  
- Hover-animaties stat cards verwijderd  
- Halve donut chart vervangen door circle chart  
- Chart labels gesorteerd: Not started → In progress → Graded → Finalized  
- Padding content verwijderd en extra onder padding toegevoegd  
- Gradient background geminimaliseerd  
- Compare grading section verkleind op desktop  
- Data ingeladen via Directus API  
- Nieuwe custom cursor geïmplementeerd  
- Cards meer contrast gegeven  
- CSS opgeschoond  
- Days left badge kleuren aangepast  

## How Has This Been Tested?
<!-- Link to test results in the Wiki-->

- [ ] User test  
- [x] Accessibility test  
- [x] Performance test  
- [x] Responsive Design test  
- [x] Device test  
- [x] Browser test  

## Images
<img width="1572" height="1040" alt="Screenshot 2025-12-17 at 00 33 46" src="https://github.com/user-attachments/assets/a0dc90ac-342c-40cd-b0b8-8afa28dad887" />
<br>
<img width="1530" height="1037" alt="Screenshot 2025-12-17 at 00 33 54" src="https://github.com/user-attachments/assets/7c02adc8-5973-4238-906c-99a6a5ccd76a" />
<!-- Usually only applicable to UI changes, what did it look like before and what will it look like after? -->

## How to review
- User test

kimnikitaschijf committed:
#176 Progress name kleur veranderen

De namen van de assessors matchen nu met de juiste progress bar kleur.

kimnikitaschijf committed:
#176 Layout naam assessor 2 netjes neerzetten

De "2" valt nu niet meer op de volgende regel maar staat netjes naast assessor.

kimnikitaschijf committed:
#176 Hover animaties weghalen bij stat cards (is overbodig en afleidend)

Hover animatie van volledige card is weggehaald. Alleen de arrow vergroot op hover en wordt lichter voor meer onderscheid.

kimnikitaschijf committed:
#176 Halve donut chart omzetten naar circle chart

Ook de kleuren labels onder chart op volgorde van kleuren in de chart gezet.

Data doormiddel van gelezen bronnen beter kunnen neerzetten voor simpelere code en betere responsiveness.

kimnikitaschijf committed:
#176 Padding content verwijderen & stat-card beter wrappen

Stat-card naar flexbox voor flex-wrap & padding van content verbeterd. Nu heb je geen lelijke grijze rand meer on t dashboard heen staan.

kimnikitaschijf committed:
#176 Gradient background minimalizeren

Nu is hij wat minder in-your-face.

kimnikitaschijf committed:
#176 Compare grading section minder groot op desktop

Heb alle items meer responsive gemaakt op t dashboard doormiddel van meer media queries, verbeterde padding, eb ook is de stat-card beter responsive in de flex

kimnikitaschijf committed:
#176 Cute custom cursor

Gewoon omdat het kan. Nieuwe custom curser ingevoegd die kan worden geactiveerd met button (sluit aan op creative coding spike week 2, maar niet exact die ivm retro stijl). Ook makkelijke ingeladen omdat hij geen custom trail nodig heeft (voor nu) en het via een img in assets wordt ingeladen.

kimnikitaschijf committed:
#176 Meer contrast tussen cards en background & header action buttons netter neergezet

Heeft nu box shadow maar niet duidelijk genoeg zichtbaar). Ook staan de header action buttons (search input & custom cursor) nu onder elkaar. Is beter responsive

kimnikitaschijf committed:
#176 Days left badge varianten, H2 & P consistente styling, continue grading section update

Days left badge bij continue grading heeft nu per aantal dagen een eigen kleur, bv groen, rood of oranje. De H2 & P's hebben 1 gezamenlijke class. Er is een view all tile toegevoegd. Complete pagina layout is beter responsive. Button components zijn goed gehandhaafd. En nog meer verbeteringen.

Ravirkt reviewed (commented):
De nieuwe styling ziet er goed uit!

Ik heb de pagina ook gecontroleerd op:
- responsiveness
- Browsertests
    - chrome
    - Firefox
    - Edge
    - Safari
    - Opera

De pagina werkt naar behoren op de verschillende browsers en is responsive.

# Custom cursor

Die custom cursor is ook heel leuk geïmplementeerd op de pagina. En ook goed dat je het aan en uit kan zetten.

Als ik Javascript uit zet dan is de cursor button nog steeds aanwezig en doet de custom cursor het niet. Je zou misschien een simpele kunnen fallback maken dat de button verdwijnt als JS uitstaat.

Voor de rest ziet alles er goed uit en is alles duidelijk aangegeven met text en kleur. 👍🏾

kimnikitaschijf committed:
#176 No JS custom cursor button fallback

Als JS uitstaat wordt de button niet meer getoond.Het wordt nu toegestaan in het button component om alle custom classes toe te staan en deze kunnen vervolgens worden aangeroepen in je NoScript. Feedback van @Ravirkt op PR.

kimnikitaschijf commented:
> De nieuwe styling ziet er goed uit!
> 
> Ik heb de pagina ook gecontroleerd op:
> 
> * responsiveness
> * Browsertests
>   
>   * chrome
>   * Firefox
>   * Edge
>   * Safari
>   * Opera
> 
> De pagina werkt naar behoren op de verschillende browsers en is responsive.
> 
> # Custom cursor
> Die custom cursor is ook heel leuk geïmplementeerd op de pagina. En ook goed dat je het aan en uit kan zetten.
> 
> Als ik Javascript uit zet dan is de cursor button nog steeds aanwezig en doet de custom cursor het niet. Je zou misschien een simpele kunnen fallback maken dat de button verdwijnt als JS uitstaat.
> 
> Voor de rest ziet alles er goed uit en is alles duidelijk aangegeven met text en kleur. 👍🏾

Super slim ja! Helemaal niet over na gedacht. Ik heb nu een fallback gemaakt die ervoor zorgt dat alle custom classes die aan de ingeladen buttons worden toegevoegd worden toegestaan. Deze kunnen vervolgens allemaal worden aangeroepen in bv `<noscript>`. Thanks voor de feedback! 🤩
